app/main.py

```
@@ -96,10 +135,302 @@ def _tc_get_json(path, params=None, timeout=30):
+    for raw in failed_step_ids:
+        if raw in step_types:
+            hits.add(step_types[raw])
+        else:
+            for sid in ids_by_len:
+                if raw.startswith(sid + "_"):
+                    hits.add(step_types[sid])
+                    break
+    return sorted(hits)
+
+
+def _check_still_failing(key, windowed_failure, step_types):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_check_still_failing`

(ANN202)